### PR TITLE
Show step by step when disabled but active

### PR DESCRIPTION
**Pull request by DilwoarH**
## What 
We want change how 'Hide navigation' option works.

At the moment when navigation is hidden its hidden for all users. We want to change this so that if a user uses the step by step navigation to get to the page with hidden navigation, the navigation shows.

## Why
Lab research has shown the value of showing step by step navigation throughout the journey.

Ticket: https://trello.com/c/SsoemgHH/860-related-to-step-by-step

# Before
screen shot 2018-09-13 at 15 31 57

# After
screen shot 2018-09-13 at 15 31 38

---

Component guide for this PR:
https://govuk-publishing-compon-pr-535.herokuapp.com/component-guide/

### CODE REVIEW

File: spec/lib/presenters/page_with_step_by_step_navigation_spec.rb [hunk 1]
```diff
@@ -345,6 +345,42 @@
...
+    end
+
+    it "does not shows related to step nav when a step by step is not active" do
+      step_nav = {
+        "content_id" => "cccc-dddd",
+        "title" => "Learn to spacewalk: small step by giant leap",
+        "base_path" => "/learn-to-spacewalk"
+      }
+      content_item = {
+        "title" => "Book a session in the vomit comet",
```
Comment (leenagupte) [inline spec/lib/presenters/page_with_step_by_step_navigation_spec.rb]: Vomit comet? 🤣

Comment (DilwoarH) [reply spec/lib/presenters/page_with_step_by_step_navigation_spec.rb]: loool copy and paste.... not my art 😆

File: lib/govuk_publishing_components/presenters/page_with_step_by_step_navigation.rb [hunk 2]
```diff
@@ -44,7 +44,8 @@ def related_links
       end
 
       def also_part_of_step_nav
-        step_by_step_navs = step_navs.delete_if { |step_nav| step_nav.content_id == active_step_by_step.content_id }
+        step_navs_list = step_navs_combined_list
```
Comment (leenagupte) [inline lib/govuk_publishing_components/presenters/page_with_step_by_step_navigation.rb]: Do we need to check the combined list for the also_part of section?

Surely, if the content_item is "part_of" (not related_to) other step_navs, they will already appear in "step_navs"?

Comment (leenagupte) [reply lib/govuk_publishing_components/presenters/page_with_step_by_step_navigation.rb]: @DilwoarH We've spoken about this in meatspace, so I'm happy for this commit to stay in.

File: spec/lib/presenters/page_with_step_by_step_navigation_spec.rb [hunk 3]
```diff
@@ -423,6 +423,55 @@
...
+        "content_id" => "aaaa-bbbb",
+        "title" => "Lose your lunch: lurch by lurch 4",
+        "base_path" => "/lose-your-lunch-4"
+      }
+      another_step_nav5 = {
+        "content_id" => "aaaa-bbbb",
+        "title" => "Lose your lunch: lurch by lurch 5",
+        "base_path" => "/lose-your-lunch-5"
+      }
+      content_item_in_two_step_navs = {
```
Comment (leenagupte) [inline spec/lib/presenters/page_with_step_by_step_navigation_spec.rb]: Nitpick: The content item is in more than two step navs.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,8 @@
 
 ## Unreleased
 
+* Show step by step when disabled but active (PR #535)
```
Comment (leenagupte) [inline CHANGELOG.md]: Perhaps expand on what "disabled but active" means?

**Pull request merged**